src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -346,7 +345,7 @@ protected final void runLeaf(Statement statement, Description description,
         EachTestNotifier eachNotifier = new EachTestNotifier(notifier, description);
         eachNotifier.fireTestStarted();
         try {
-            statement.evaluate();
```

Comment:
I wonder if instead of updating `runLeaf()` we should update `methodBlock()` to return a `Statement` that calls `Thread.interrupted()` at the end of the statement chain.

Reply:
We should almost certainly modify `classBlock()` to return a `Statement` that calls T`hread.interrupted()` at the end of the statement chain.

Reply:
@kcooney Updated `methodBlock()` to return a statement that clears interrupt status and reverted `runLeaf`. I am not sure how updating `classBlock()` will help?

Reply:
@npathai I assume that we would want to handle the case where the interrupt bit was set  in a class rule or `@AfterClass` method. Those are handled by `classBlock()`